### DISCUSSION TITLE
Sympy physics mechanics. Sequence of variables

### DISCUSSION THREAD

**Discussion by Peter230655** [sympy/sympy#21736] [Q&A] [answered]
I want to establish Kane's equation of motion with, say,
generalized coordinates: q = ( q1, q2, q3, q4)
independent speeds        u = ( u1, u4)                 ( where ui = d/dt(qi), i = 1,2,3, 4 )
dependent speeds           u_dep = (u2, u3)

I would set it up like this:
KM = KanesMethod(N, q_ind=q, u_ind=u, u_dependent=u_dep, kd_eqs=kd, velocitiy_constraints=speed_constr)

What is the sequence of the coordinates in rhs = KM.rhs() ?
Is it
d/dt(q1, q2, q3, q4, u1, u2, u3, u4),      or
d/dt(q1, q2 ,q3, q4, u1, u4, u2, u3),      or
still different?

Thanks a lot for any help!

**Comment by moorepants** [comment, marked as answer]
```python
KM.q
KM.u
```

should give you the coordinates and speeds in the order used internally to `KanesMethod`. You can also see that the independent are stack on top of the dependent coordinates and speeds here: https://docs.sympy.org/latest/modules/physics/mechanics/kane.html

**Comment by Peter230655** [comment]
KM.q and KM.u worked perfect!
Thanks!

On Mon 12. Jul 2021 at 15:40, Jason K. Moore ***@***.***>
wrote:

> KM.qKM.u
>
> should give you the coordinates and speeds in the order used internally to
> KanesMethod. You can also see that the independent are stack on top of
> the dependent coordinates and speeds here:
> https://docs.sympy.org/latest/modules/physics/mechanics/kane.html
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/sympy/sympy/discussions/21736#discussioncomment-993753>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT5MQUQAA3NF5CPNFX6U52DTXLWGBANCNFSM5AGYXQNA>
> .
>
-- 
Best regards,

Peter Stahlecker
